l10n_ch_invoice_reports/models/report.py

```diff
@@ -45,26 +45,26 @@ def render_qweb_pdf(self, res_ids=None, data=None):
         ]
         if self.report_name not in reports or not res_ids:
             return super().render_qweb_pdf(res_ids, data)
-        inv_report = self._get_report_from_name("account.report_invoice")
-        invoice_pdf, _ = inv_report.render_qweb_pdf(res_ids, data)
-        invoice_pdf_io = io.BytesIO(invoice_pdf)
+        tmp_stream_io = []
+        for res_id in res_ids:
+            inv_report = self._get_report_from_name("account.report_invoice")
+            invoice_pdf, _ = inv_report.render_qweb_pdf(res_id, data)
+            invoice_pdf_io = io.BytesIO(invoice_pdf)
 
-        isr_report = self._get_report_from_name("l10n_ch.isr_report_main")
-        isr_pdf, _ = isr_report.render_qweb_pdf(res_ids, data)
-        isr_pdf_io = io.BytesIO(isr_pdf)
+            isr_report = self._get_report_from_name("l10n_ch.isr_report_main")
+            isr_pdf, _ = isr_report.render_qweb_pdf(res_id, data)
```

Comment:
Are we computing ISR and QR report everytime we call the merge? Even when we don't use them?

Seems we could check `self.report_name` before generating both.
(Not directly related to this PR though.